nlisim/modules/state_output.py

```diff
@@ -40,8 +40,14 @@ def initialize(self, state: State) -> State:
         output_dir = self._output_dir
         if output_dir.exists():
             print(f'File output directory {output_dir.resolve()} exists. Clearing it.')
-            shutil.rmtree(output_dir)
-        output_dir.mkdir(parents=True)
+            for file in output_dir.iterdir():
```

Comment:
Maybe this is for debugging, but if it ends up being necessary, we should document why via a code comment or bug report.

Reply:
This is because the output folder is mounted as a shared volume. So this folder itself can't be removed, instead, we should remove the content within the output folder. I will add a comment on it later.

Reply:
I've pushed a commit to fix this. It's also cleaner to put the logic in a separate method, as opposed to inline.